nx/source/services/acc.c

```diff
@@ -33,6 +34,113 @@ Service* accountGetService(void) {
     return &g_accSrv;
 }
 
+Result accountGetUserCount(s32* user_count)
+{
+    IpcCommand c;
+    ipcInitialize(&c);
+
+    struct {
+        u64 magic;
+        u64 cmd_id;
+    } *raw;
+
+    raw = ipcPrepareHeader(&c, sizeof(*raw));
+
+    raw->magic = SFCI_MAGIC;
+    raw->cmd_id = 0;
+
+    Result rc = serviceIpcDispatch(&g_accSrv);
+
+    if (R_SUCCEEDED(rc)) {
+        IpcParsedCommand r;
+        ipcParse(&r);
+
+        struct {
+            u64 magic;
+            u64 result;
+            u32 user_count;
+        } *resp = r.Raw;
+
+        rc = resp->result;
+
+        if (R_SUCCEEDED(rc)) {
+            *user_count = resp->user_count;
+        }
+    }
+
+    return rc;
+}
+
+static Result _accountListAllUsers(u128* userIDs)
+{
+    IpcCommand c;
+    ipcInitialize(&c);
+
+    Result rc=0;
+
+    size_t bufsize = ACC_USER_LIST_SIZE * sizeof(u128);
+
+    ipcAddRecvStatic(&c, userIDs, bufsize, 0);
+
+    struct {
+        u64 magic;
+        u64 cmd_id;
+    } *raw;
+
+    raw = ipcPrepareHeader(&c, sizeof(*raw));
+
+    raw->magic = SFCI_MAGIC;
+    raw->cmd_id = 2;
+
+    rc = serviceIpcDispatch(&g_accSrv);
+
+    if (R_SUCCEEDED(rc)) {
+        IpcParsedCommand r;
+        ipcParse(&r);
+
+        struct {
+            u64 magic;
+            u64 result;
+        } *resp = r.Raw;
+
+        rc = resp->result;
+    }
+
+    return rc;
+}
+
+Result accountListAllUsers(u128* userIDs, size_t max_userIDs, size_t *actual_total)
+{
+    Result rc=0;
+
+    u128 *temp_userIDs;
+	temp_userIDs = malloc(sizeof(u128) * ACC_USER_LIST_SIZE);
+
+    rc = _accountListAllUsers(temp_userIDs);
+
+    if (R_SUCCEEDED(rc)) {
+        size_t total_userIDs = 0;
+
+        for (int i = 0; i < ACC_USER_LIST_SIZE; i++) {
+            if (temp_userIDs[i]) {
+                total_userIDs++;
+            }
+        }
+
+        if (max_userIDs > total_userIDs) {
+            max_userIDs = total_userIDs;
+        }
+
+        for (int i = 0; i < max_userIDs; i++) {
+            userIDs[i] = temp_userIDs[i];
+        }
+
+        *actual_total = total_userIDs;
```

Comment:
This is cumbersome. *actual_total should be assigned max_userIDs, which is the number of users that were returned.

Reply:
Happy to change it, but this contradicts the original comment which said:
"Where actual_total is calculated by how many userIDs are non-zero, stopping at the first all-zero one."
The number which are non-zero is likely to be different to the max_userIDs.
Please confirm which way you'd prefer.

Reply:
Yeah, that makes sense. I've changed it to be as you suggested now.

Reply:
The original statement is sort of ambiguous, but it's a common API convention to do it like what I said.